packages/cactus-plugin-persistence-ethereum/README.md

```diff
@@ -0,0 +1,237 @@
+# `@hyperledger/cactus-plugin-persistence-ethereum`
+
+This plugin allows `Cactus` to persist Ethereum data into some storage (currently to a `PostgreSQL` database, but this concept can be extended further).
+Data in the database can later be analyzed and viewed in a GUI tool.
+
+## Summary
+
+- [Remarks](#remarks)
+- [Getting Started](#getting-started)
+- [Endpoints](#endpoints)
+- [Running the tests](#running-the-tests)
+- [Contributing](#contributing)
+- [License](#license)
+- [Acknowledgments](#acknowledgments)
+
+## Remarks
+
+- This plugin was only tested with small, permissioned Ethereum ledgers. Running it to archive and monitor large ledgers (like main net) is not recommended yet.
+- For now, the database schema is not considered public and can change over time (i.e., writing own application that reads data directly from the database is discouraged).
+- Only `status` endpoint is available, all the methods must be called directly on the plugin instance for now.
+- Monitored ERC20 tokens should be added before synchronizing the database (previous transfers will not be parsed correctly if you later add the token).
+
+## Getting Started
+
+Clone the git repository on your local machine. Follow these instructions that will get you a copy of the project up and running on your local machine for development and testing purposes.
+
+### Prerequisites
+
+In the root of the project, execute the command to install and build the dependencies. It will also build this persistence plugin:
+
+```sh
+yarn run configure
+```
+
+### Usage
+
+Instantiate a new `PluginPersistenceEthereum` instance:
+
+```typescript
+import { PluginPersistenceEthereum } from "@hyperledger/cactus-plugin-persistence-ethereum";
+import { v4 as uuidv4 } from "uuid";
+
+const persistencePlugin = new PluginPersistenceEthereum({
+  instanceId: uuidv4(),
+  apiClient: new SocketIOApiClient(apiConfigOptions),
+  logLevel: "info",
+  connectionString:
+    "postgresql://postgres:your-super-secret-and-long-postgres-password@localhost:5432/postgres",
+});
+
+// Initialize the connection to the DB
+persistencePlugin.onPluginInit();
+```
+
+Alternatively, import `PluginFactoryLedgerPersistence` from the plugin package and use it to create a plugin.
+
+```typescript
+import { PluginFactoryLedgerPersistence } from "@hyperledger/cactus-plugin-persistence-ethereum";
+import { PluginImportType } from "@hyperledger/cactus-core-api";
+import { v4 as uuidv4 } from "uuid";
+
+const factory = new PluginFactoryLedgerPersistence({
+  pluginImportType: PluginImportType.Local,
+});
+
+const persistencePlugin = await factory.create({
+  instanceId: uuidv4(),
+  apiClient: new SocketIOApiClient(apiConfigOptions),
+  logLevel: "info",
+  connectionString:
+    "postgresql://postgres:your-super-secret-and-long-postgres-password@localhost:5432/postgres",
+});
+
+// Initialize the connection to the DB
+persistencePlugin.onPluginInit();
```

Comment:
@outSH `onPluginInit()` is an async call so I'd make sure to `await` on it in the examples otherwise people using this code as inspiration might get race conditions in their applications where they start initializing the plugin but then immediately try to do something else that depends on the plugin having finished initialization.

Reply:
@petermetz Fixed.